openmc/tally_derivative.py

```diff
@@ -0,0 +1,113 @@
+import sys
+from numbers import Integral
+from xml.etree import ElementTree as ET
+
+import openmc.checkvalue as cv
+from openmc.mixin import EqualityMixin, IDManagerMixin
+
+
+class TallyDerivative(EqualityMixin, IDManagerMixin):
+    """A material perturbation derivative to apply to a tally.
+
+    Parameters
+    ----------
+    derivative_id : int, optional
+        Unique identifier for the tally derivative. If none is specified, an
+        identifier will automatically be assigned
+    variable : str, optional
+        Accepted values are 'density', 'nuclide_density', and 'temperature'
+    material : int, optional
+        The perturbed material ID
+    nuclide : str, optional
+        The perturbed nuclide. Only needed for 'nuclide_density' derivatives.
+        Ex: 'Xe135'
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the tally derivative
+    variable : str
+        Accepted values are 'density', 'nuclide_density', and 'temperature'
+    material : int
+        The perturubed material ID
+    nuclide : str
+        The perturbed nuclide. Only needed for 'nuclide_density' derivatives.
+        Ex: 'Xe135'
+
+    """
+
+    next_id = 1
+    used_ids = set()
+
+    def __init__(self, derivative_id=None, variable=None, material=None,
+                 nuclide=None):
+        # Initialize Tally class attributes
+        self.id = derivative_id
+        self.variable = variable
+        self.material = material
+        self.nuclide = nuclide
+
+    def __repr__(self):
+        string = 'Tally Derivative\n'
+        string += '{: <16}=\t{}\n'.format('\tID', self.id)
+        string += '{: <16}=\t{}\n'.format('\tVariable', self.variable)
+
+        if self.variable == 'density':
+            string += '{: <16}=\t{}\n'.format('\tMaterial', self.material)
+        elif self.variable == 'nuclide_density':
+            string += '{: <16}=\t{}\n'.format('\tMaterial', self.material)
+            string += '{: <16}=\t{}\n'.format('\tNuclide', self.nuclide)
+        elif self.variable == 'temperature':
+            string += '{: <16}=\t{}\n'.format('\tMaterial', self.material)
```

Comment:
you are setting same value in each if condition, it can be simplified by:
```
if self.variable in ['density', 'nuclide_density', 'temperature']:
  string += '{: <16}=\t{}\n'.format('\tMaterial', self.material)
if self.variable == 'nuclide_density':
  string += '{: <16}=\t{}\n'.format('\tNuclide', self.nuclide)
```